### PR TITLE
Add git workflow to auto gen docs on master push

### PR DESCRIPTION
**Pull request by jeremytchang**
Npm run docs should be called everytime there's a change
to our source or README, or else the docs will be outdated.

Added a git workflow to auto gen docs on pushes to master
so that manual commits are cleaner with only necessary changes,
and auto commits can contain the auto generated doc changes that
we should not worry about.

### CODE REVIEW

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+name: Auto Doc Gen
+on:
+  push:
+    branches:
+      - master
+
+env:
+  NODE_VERSION: 14.x
```
Comment (jkaster) [inline .github/workflows/docs.yml:8]: not super important but is Node 16 available for this?

**Review by jkaster — APPROVED** [review]
Looks like it will work to me

File: .github/workflows/docs.yml (hunk 1 above)
Comment (jeremytchang) [reply .github/workflows/docs.yml:8]: I'll use 16.x. Was using the latest one defined on the other workflow, but Bryn just updated it to 16.x.

**Comment by jeremytchang** [comment]
Yup, it should work. I've tested this locally already outside of the `git push`.

File: .github/workflows/docs.yml (hunk 1 above)
Comment (jkaster) [reply .github/workflows/docs.yml:8]: Thanks!

**Review by jkaster — APPROVED** [review]
Nice!

**Comment by jkaster** [comment]
There is another ticket to update the publish workflow to make it use the same automation hooks as SDK codegen

**Comment by jeremytchang** [comment]
@bryans99 The workflow will not commit/push any changes if `npm run docs` does not change any files. Also the dependabot upgrade pulled in a breaking change for typedocs. I made a ticket to fix that.

**Comment by bryans99** [comment]
I think we should consider rolling back the version of typedoc (at least for now). I noticed a number of differences that I dont yet have the time to go into. As long as it does not introduce dependabot issues I would say we should be good to do that

**Pull request merged**